### ISSUE TITLE
Uncaught TypeError: Patch does not apply

### ISSUE THREAD

**Issue by abi-ogawa** [atom/atom#15480]
[Enter steps to reproduce:]

1. ...
2. ...

**Atom**: 1.19.4 ia32
**Electron**: 1.6.9
**OS**: Unknown Windows version
**Thrown From**: Atom Core

### Stack Trace

Uncaught TypeError: Patch does not apply

```
At C:\Users\Ogawa\AppData\Local\atom\app-1.19.4\resources\app\node_modules\text-buffer\lib\history.js:144

TypeError: Patch does not apply
    at History.module.exports.History.getChangesSinceCheckpoint (~/AppData/Local/atom/app-1.19.4/resources/app/node_modules/text-buffer/lib/history.js:144:34)
    at TextBuffer.module.exports.TextBuffer.getChangesSinceCheckpoint (~/AppData/Local/atom/app-1.19.4/resources/app/node_modules/text-buffer/lib/text-buffer.js:964:38)
    at TextBuffer.module.exports.TextBuffer.emitDidStopChangingEvent (~/AppData/Local/atom/app-1.19.4/resources/app/node_modules/text-buffer/lib/text-buffer.js:1579:36)
    at later (~/AppData/Local/atom/app-1.19.4/resources/app/node_modules/text-buffer/lib/helpers.js:21:13)
```

### Commands

```
     -1:04.4.0 core:backspace (input.hidden-input)
     -1:02.8.0 intentions:highlight (input.hidden-input)
  5x -1:02.6.0 core:save (input.hidden-input)
     -0:55.9.0 intentions:highlight (input.hidden-input)
     -0:55.7.0 core:save (input.hidden-input)
     -0:53.2.0 intentions:highlight (input.hidden-input)
  5x -0:53 core:save (input.hidden-input)
     -0:47.1.0 intentions:highlight (input.hidden-input)
  6x -0:46.9.0 core:save (input.hidden-input)
     -0:18.3.0 intentions:highlight (input.hidden-input)
  2x -0:18.1.0 core:copy (input.hidden-input)
     -0:17.6.0 core:backspace (input.hidden-input)
     -0:15.5.0 emmet:expand-abbreviation-with-tab (input.hidden-input)
     -0:15.1.0 intentions:highlight (input.hidden-input)
     -0:15 core:paste (input.hidden-input)
  3x -0:14.6.0 core:save (input.hidden-input)
```

### Non-Core Packages

```
activate-power-mode 2.4.0 
atom-beautify 0.30.5 
atom-clock 0.1.13 
atom-material-syntax 1.0.6 
atom-material-syntax-dark 1.0.0 
atom-material-syntax-light 0.4.6 
atom-material-ui 2.0.4 
aurora-theme 1.2.2 
autocomplete-paths 2.8.1 
busy-signal 1.4.3 
color-picker 2.2.5 
emmet 2.4.3 
emmet-snippets-compatibility 1.0.6 
file-icons 2.1.11 
git-log 0.4.1 
git-plus 7.9.3 
highlight-selected 0.13.1 
intentions 1.1.5 
language-actionscript3 0.3.0 
linter 2.2.0 
linter-coffeelint 1.3.1 
linter-jshint 3.1.5 
linter-ui-default 1.6.8 
merge-conflicts 1.4.5 
minimap 4.29.6 
minimap-bookmarks 0.4.2 
minimap-cursorline 0.2.0 
minimap-find-and-replace 4.5.2 
minimap-git-diff 4.3.1 
minimap-highlight-selected 4.6.1 
minimap-pigments 0.2.2 
minimap-selection 4.5.0 
monokai 0.24.0 
pigments 0.40.2 
pretty-json 1.6.4 
project-manager 3.3.5 
sync-settings 0.8.2 
tabs-to-spaces 1.0.3 
tag 0.5.0 
tree-view-git-status 1.4.0 
turbo-javascript 1.4.0 
vim-mode 0.66.0 
```

**Comment by Ben3eeE** [comment]
Thanks for the report!  Can you confirm a few things for us?

* what were you doing at the time of the error?
* is the error is reproducible?
* if reproducible, do you see the error in safe mode (atom --safe)?

**Comment by abi-ogawa** [comment]
Thank you for your reply! I will answer the question.

・what were you doing at the time of the error?
.php file edit

・is the error is reproducible?
yes.
It happens every time you edit a .php file.

・if reproducible, do you see the error in safe mode (atom --safe)?
yes.
The same error occurs.

**Comment by Ben3eeE** [comment]
@abi-ogawa Does this only happen with one php file or with all php files you edit? Can you share a file where you see this error?

/cc: @maxbrunsfeld

**Comment by abi-ogawa** [comment]
sorry...
I fixed it after you reported it,
No error occurs.

**Comment by Ben3eeE** [comment]
@abi-ogawa Can you please tell how you fixed it? Thank you :bowing_man:

**Comment by piotr-cz** [comment]
This error pops up on every keystroke along with strange noise effect (some lines are not updated after scroll) inside syntax editor, when editing particular files after upgrade to _v1.9.4_.
When I close the file and reopen again, error doesn't show up. It's quite hard to reproduce but happened for me when editing _jsx_ files.

Stack trace same as in issue report

**Comment by maxbrunsfeld** [comment]
@piotr-cz @abi-ogawa Does this happen if you run Atom with `atom --safe`?

@piotr-cz Could you share the list of packages that you have installed?

**Comment by piotr-cz** [comment]
@maxbrunsfeld 

The issue comes up after starting Atom and only on focused tab.
It goes away when I close the tab and reopen file again until I close and open Atom again.

After starting atom with `--safe` flag the issue was there, but after closing Atom it seems it's gone and editor starts faster

[Enter steps to reproduce:]

1. Open Atom editor
2. Edit currently opened file

**Atom**: 1.19.4 x64
**Electron**: 1.6.9
**OS**: Microsoft Windows 10 Home
**Thrown From**: Atom Core


### Stack Trace

Uncaught TypeError: Patch does not apply

```
At C:\Users\Piotr\AppData\Local\atom\app-1.19.4\resources\app\node_modules\text-buffer\lib\history.js:144

TypeError: Patch does not apply
    at History.module.exports.History.getChangesSinceCheckpoint (~/AppData/Local/atom/app-1.19.4/resources/app/node_modules/text-buffer/lib/history.js:144:34)
    at TextBuffer.module.exports.TextBuffer.getChangesSinceCheckpoint (~/AppData/Local/atom/app-1.19.4/resources/app/node_modules/text-buffer/lib/text-buffer.js:964:38)
    at TextBuffer.module.exports.TextBuffer.emitDidStopChangingEvent (~/AppData/Local/atom/app-1.19.4/resources/app/node_modules/text-buffer/lib/text-buffer.js:1579:36)
    at later (~/AppData/Local/atom/app-1.19.4/resources/app/node_modules/text-buffer/lib/helpers.js:21:13)
```

### Commands

```
     -4:00.2.0 editor:move-to-end-of-screen-line (input.hidden-input)
     -3:58.8.0 editor:newline (input.hidden-input)
     -3:58.8.0 intentions:highlight (input.hidden-input)
     -3:58.6.0 core:paste (input.hidden-input)
     -3:58.2.0 core:save (input.hidden-input)
     -3:51.9.0 intentions:highlight (input.hidden-input)
     -3:51.8.0 core:save (input.hidden-input)
     -3:46.9.0 intentions:highlight (input.hidden-input)
     -3:46.8.0 core:save (input.hidden-input)
  3x -3:44.3.0 core:move-right (input.hidden-input)
     -3:42.2.0 core:move-down (input.hidden-input)
     -3:41.9.0 core:backspace (input.hidden-input)
     -3:41.7.0 intentions:highlight (input.hidden-input)
     -3:41.6.0 core:save (input.hidden-input)
     -3:37.4.0 intentions:highlight (input.hidden-input)
     -3:37.3.0 core:save (input.hidden-input)
```

### Non-Core Packages

```
atom-ternjs 0.18.3 
autoclose-html 0.23.0 
autocomplete-modules 1.7.0 
busy-signal 1.4.3 
ctrl-dir-scroll 0.2.5 
editorconfig 2.2.2 
file-icons 2.1.11 
flex-tool-bar 0.12.0 
git-diff-details 1.4.0 
highlight-selected 0.13.1 
hyperclick 0.1.3 
intentions 1.1.5 
js-hyperclick 1.11.0 
language-twig 1.6.3 
linter 2.2.0 
linter-eslint 8.2.1 
linter-jscs undefined 
linter-jshint 3.1.5 
linter-stylelint 4.0.2 
linter-ui-default 1.6.8 
minimap 4.29.6 
minimap-autohide 0.10.1 
move-status-items 0.0.11 
php-debug 0.2.5 
php-hyperclick 1.2.1 
project-manager 3.3.5 
react 0.16.2 
tool-bar 1.1.0 
tool-bar-main 0.0.10 
zenburn 3.1.0 
```

**Comment by Ben3eeE** [comment]
@piotr-cz Can you confirm if the issue comes back if you start up Atom outside of safe mode again?

**Comment by jkd77** [comment]
I have the same issue and it is rough. Also happens in safe mode.

**Comment by Ben3eeE** [comment]
@jkd77 Please post the output of `apm list --installed`

**Comment by jkd77** [comment]
apm list --installed
Community Packages (3) /Users/paulschlereth/.atom/packages
├── language-cfml@0.17.1
├── minimap@4.29.6
└── sass-autocompile@0.13.3 (disabled)

**Comment by jkd77** [comment]
same error occurs even after disabling all Community packages

**Comment by Ben3eeE** [comment]
@jkd77 Is there any more information you can provide?

* Which file type are you editing?
* Do you see it in every file or just one file?
* What operating system are you using?
* Do you see any other errors? Any additional information in the developer tools console?
* Do you still see the error if you close and reopen the file?

**Comment by jkd77** [comment]
osx sierra 10.12.6, editing scss and html files, I'm trying to get the complete error message but it hasn't appeared after restarting. I'll post back when it does.

**Comment by theorosendorf** [comment]
[Enter steps to reproduce:]

1. ...
2. ...

**Atom**: 1.19.4 x64
**Electron**: 1.6.9
**OS**: Mac OS X 10.12.6
**Thrown From**: Atom Core


### Stack Trace

Uncaught TypeError: Patch does not apply

```
At /Applications/Atom.app/Contents/Resources/app/node_modules/text-buffer/lib/history.js:144

TypeError: Patch does not apply
    at History.module.exports.History.getChangesSinceCheckpoint (/Applications/Atom.app/Contents/Resources/app/node_modules/text-buffer/lib/history.js:144:34)
    at TextBuffer.module.exports.TextBuffer.getChangesSinceCheckpoint (/Applications/Atom.app/Contents/Resources/app/node_modules/text-buffer/lib/text-buffer.js:964:38)
    at TextBuffer.module.exports.TextBuffer.emitDidStopChangingEvent (/Applications/Atom.app/Contents/Resources/app/node_modules/text-buffer/lib/text-buffer.js:1579:36)
    at later (/Applications/Atom.app/Contents/Resources/app/node_modules/text-buffer/lib/helpers.js:21:13)
```

### Commands

```
     -0:02.7.0 core:backspace (input.hidden-input)
```

### Non-Core Packages

```
atom-panels 0.4.1 
busy-signal 1.4.3 
chester-atom-syntax 0.3.0 
compare-files 0.8.1 
emmet 2.4.3 
git-log 0.4.1 
intentions 1.1.5 
linter 2.2.0 
linter-csslint 2.0.0 
linter-ui-default 1.6.8 
minimap-split-diff 0.3.7 
pigments 0.40.2 
split-diff 1.5.1 
```

**Comment by piotr-cz** [comment]
After two days error didn't come back even once.
Looks like starting Atom with `--safe` flag helped, now I'm starting Atom without a flag.

**Comment by maxbrunsfeld** [comment]
This may be fixed in Atom 1.19.5 via https://github.com/atom/text-buffer/pull/252. Everyone, please let me know if this continues to be a problem in Atom 1.19.5.
